design/grammar.md

```diff
@@ -0,0 +1,13 @@
+# SeedBlock and SeedX Grammars
+
+## SeedBlock
+
+The syntax of SeedBlock is defined by an ANTLR v4 grammar file named [SeedBlock.g4](/grammars/SeedBlock.g4). The parser generated by ANTLR tool is used to parse the whole SeedBlock program or validate the partial user input during inline editing.
+
+The SeedBlock program is serialized into the memory with JSON Block Exchange Format, which includes the information of the blocks UI and the parsing tree. It's the main storage format of SeedBlock and can be shared with other users. A Visitor is used to walk through the parsing tree to transfer the SeedBlock program into SeedAST.
+
+## SeedX Languages
+
+The syntax of SeedX languages are defined by the ANTLR v4 grammar files located in the grammars directory. The parser and visitor generated by ANTLR tool are used to walk through the parsing tree to transfer the SeedX program into SeedAST.
```

Comment:
The syntax of SeedX languages are -> The syntax of SeedX languages is

in the grammars directory -> in the `grammars` directory

Reply:
ANTLR tool -> the ANTLR tool

Reply:
Done